treemacs-impl.el

```diff
@@ -684,15 +684,18 @@ treemacs buffer teardown\) otherwise the currently selected frame is used."
 (defun treemacs--setup-buffer ()
   "Create and setup a buffer for treemacs in the right position and size."
   (treemacs--forget-last-highlight)
-  (-> (selected-window)
-      (frame-root-window)
-      (split-window nil treemacs-position)
-      (select-window))
-  (let ((buf (treemacs--get-framelocal-buffer)))
-    (switch-to-buffer buf)
-    (bury-buffer buf))
-  (set-window-dedicated-p (selected-window) t)
-  (let ((window-size-fixed))
+  (let* ((buf (treemacs--get-framelocal-buffer))
+	 (win (select-window (display-buffer-in-side-window
```

Comment:
No side windows, at least not without an option. 1) They are unsplittable, which may leads to conflicts with packages like shell-pop. 2) The window parameters that define a side window have been changed with emacs26, so that would have to be taken into account as well.

Best just stick with slitting the frame root.